### PR TITLE
feat(cb2-7058)/feat(cb2-7059): Generate C and Z numbers

### PR DESCRIPTION
**Pull request by mcadenas-bjss**
## Generate C and Z numbers

[CB2-7058](https://dvsa.atlassian.net/browse/CB2-7058)
[CB2-7059](https://dvsa.atlassian.net/browse/CB2-7059)

## Checklist

- [ ] Code has been tested manually
- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/domain/Processors/TrailerProcessor.ts [hunk 1]
```diff
@@ -46,8 +45,12 @@ export class TrailerProcessor extends VehicleProcessor<Trailer> {
 
   /* #region  Async functions */
   protected async setNumberKey(): Promise<void> {
-      this.vehicle.systemNumber = await this.numberGenerator.generateSystemNumber();
-      this.vehicle.trailerId = await this.numberGenerator.generateTrailerId();
+    this.vehicle.systemNumber = await this.numberGenerator.generateSystemNumber();
+    if(!this.vehicle.trailerId) {
+      const newTrailerId = await this.numberGenerator.generateTrailerId();
+      this.vehicle.trailerId = newTrailerId;
+      this.vehicle.primaryVrm = newTrailerId;
```
Comment (gjulien-bjss) [inline src/domain/Processors/TrailerProcessor.ts:52]: Not too sure on the Vrm on trl

File: src/models/TechRecordsDAO.ts [hunk 2]
```diff
@@ -181,6 +185,15 @@ class TechRecordsDAO {
           resource: "/system-number/",
         })
 
+  public getZNumber = () =>
```
Comment (gjulien-bjss) [inline src/models/TechRecordsDAO.ts:188]: I'm not sure if there are some unit tests for the trailiD which can be copied over

File: src/handlers/NumberGenerator.ts [hunk 3]
```diff
@@ -38,4 +38,19 @@ export class NumberGenerator {
       return Promise.reject({statusCode: 500, body: error});
     }
   }
+
+  public async generateZNumber(): Promise<string> {
```
Comment (gjulien-bjss) [inline src/handlers/NumberGenerator.ts:42]: I'm not sure if there are some unit tests for the trailerId which can be copied over

File: tests/unit/TechRecordsService.unitTest.ts [hunk 4]
```diff
@@ -282,6 +285,35 @@ describe("insertTechRecord", () => {
...
+              });
+            },
+            getSystemNumber: () => Promise.resolve({systemNumber: "10000001", testNumberKey: 3}),
+            getZNumber: () => Promise.resolve({zNumber: "1000001Z"}),
+          };
+        });
+        const techRecordsService = new TechRecordsService(new MockDAO());
+
+        const data = await techRecordsService.insertTechRecord(techRecord, msUserDetails);
+        console.log("🚀 ~ file: TechRecordsService.unitTest.ts:337 ~ it ~ data", data);
```
Comment (gjulien-bjss) [inline tests/unit/TechRecordsService.unitTest.ts:313]:
```suggestion
```

**Review by tom-bjss — APPROVED** [review]
I dont know about the trailed ID populating the VRM field

**Comment by gjulien-bjss** [comment]
> I dont know about the trailed ID populating the VRM field

Something to do about the history of trailer Id, historic trailer Ids are stored in the secondaryVrms, because... well you know....

**Review by Ruairidh-BJSS — APPROVED** [review]

**Pull request merged**